hnn_core/network.py

```diff
@@ -405,6 +408,40 @@ def __eq__(self, other):
                 self._gids == other._gids and
                 self._types == other._types)
 
+    def __getitem__(self, gid_item):
```

Comment:
can you add a docstring above to understand what it does? Just for inspiration:

https://github.com/mne-tools/mne-python/blob/270b77768e4d2cf0f3f7ae4defd229c9f39eb2af/mne/utils/mixin.py#L93-L126

But don't get this fancy. [Keep it simple stupid](https://en.wikipedia.org/wiki/KISS_principle). MNE slicing functionality has expanded over 10 years ... it started simple.

I'm thinking that you want to be index by gid (and maybe by trial index)

Reply:
another alternative is that `Spikes` object will store only info for a single cell and then you store a list of spike objects. In that case `plot_spikes` will probably have to be a function that takes the list of spikes as input rather than a method of `spikes` class ... I kind of like this alternative better because then you can also in the future store cell geometry information in this class. Think of the pros and cons, we'll probably have to discuss IRL but so far I'm leaning towards this second alternative (depending on how easy it is to implement, don't want things to get complicated, cf KISS principle)

Reply:
Storing it as a list would be less cumbersome in terms of updating information stored. As you can see a lot of code is dedicated to just reformatting the data after filtering. However it sounds like that change will require a lot of changes outside the `Spikes` class.

I'll look to characterize how wide reaching that change will be with respect to the current code. My motivation for the current change is it has zero impact on the rest of the code-base. I'll stick with this implementation while I improve how voltage recordings are passed around.

Reply:
perhaps storing `Spikes` as a list could be the next PR ... we do want to have clean data structures before making a release though.

Reply:
I really like the idea of a new `CellResponse` (or whatever we want to call it) object that stores simulation output data like spikes, soma voltage, etc. on a cell-by-cell basis. We can defs save this for a different PR though.